### DISCUSSION TITLE
Do we need to run VAD before running pyannote - speaker diarization pipeline ?

### DISCUSSION THREAD

**Discussion by amitli1** [pyannote/pyannote-audio#1295] [Development]
According to [pyannote.audio: neural building blocks for speaker diarization](https://arxiv.org/abs/1911.01255) (pyannote version 1) the pipeline **includes** VAD blocks.

According to pyannote version 2.1:
 [PYANNOTE.AUDIO 2.1 SPEAKER DIARIZATION PIPELINE: PRINCIPLE, BENCHMARK, AND RECIPE](https://huggingface.co/bhuvanesh25/pyannote-diar-copy/resolve/main/technical_report_2.1.pdf)

they overhauled the  speaker diarization pipeline and removed some blocks (including VAD, overlapped speech detection).

So if I'm using pyannote - speaker diarization pipeline (version 2.1.1),
Do I need to run VAD and cut the speaking segments before running the  speaker diarization pipeline  ?

**Comment by hbredin** [comment]
No, you do not need to run VAD before.

Also, even if you do have your own VAD, there is no easy way to pass it to the pipeline.

**Reply by amitli1** [reply]
So segments without speech will be concentrated to one cluster ("no speech cluster") ?

**Reply by hbredin** [reply]
No. There will simply be gaps in the output for non-speech regions.
